packages/shorebird_cli/test/src/shorebird_env_test.dart

```diff
@@ -717,5 +717,111 @@ base_url: https://example.com''');
         expect(runWithOverrides(() => shorebirdEnv.isRunningOnCI), isFalse);
       });
     });
+
+    group('addShorebirdYamlToPubspecAssets', () {
+      const pubspecContents = '''
+name: test
+version: 1.0.0
+environment:
+ sdk: ">=2.19.0 <3.0.0"''';
+      test('creates flutter.assets and adds shorebird.yaml', () {
+        final tempDir = Directory.systemTemp.createTempSync();
+        final pubspecFile = File(p.join(tempDir.path, 'pubspec.yaml'))
+          ..createSync()
+          ..writeAsStringSync(pubspecContents);
+        IOOverrides.runZoned(
+          () => runWithOverrides(
+            () => shorebirdEnv.addShorebirdYamlToPubspecAssets(),
```

Comment:
I think this can be a tear-off (ex: `runWithOverrides(shorebirdEnv.addShorebirdYamlToPubspecAssets)`)